.github/workflows/client.yml

```diff
@@ -0,0 +1,30 @@
+name: Go
+
+#TODO: extend the conditions once workflow gets tested together with other workflows 
+on: [push, pull_request]
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true # ignore the failure of a step and avoid terminating the job
+      - name: Create test results directory
+        run: |
+          mkdir -p /tmp/test-results/keep-core-go
+      - name: Run Docker build
+        run: |
+          docker build --target gobuild --tag go-build-env .
+          docker build --tag keep-client . 
```

Comment:
Maybe we could switch places for these two? First run docker build and then create test results directory? We don't need to create the dir for running the build but for the following testing steps.